### PR TITLE
[llvm][docs] Document how to get admin permissions for a Buildbot worker

### PR DESCRIPTION
**Pull request by DavidSpickett**
I spent a long time trying different combinations of primary and verified emails, until a colleague tried it who happened to have a public profile email set when I did not.

Document how this works to save everyone else the legwork.

### CODE REVIEW

**Comment by DavidSpickett** [comment]
And if you're wondering, the console output looks like:
```
{"jsonrpc":"2.0","id":332,"error":{"code":-32600,"message":" You must be the worker owner."}}
<as many times as there are pending requests>
```
I'm not sure why I thought to check it at all, so I figured it was worth mentioning even though it's a bit of an unintended feature.

File: llvm/docs/HowToAddABuilder.rst [hunk 1]
```diff
@@ -276,4 +276,32 @@ Leave it on the staging buildmaster
...
+  restart the worker. You should see the new admin details in the web interface
+  shortly afterwards.
+
+* Connect GitHub to Buildbot by clicking on the "Anonymous" button on the
+  top right of the page, then "Login with GitHub" and authorise the app.
+
+Some tasks don't give immediate feedback, so if nothing happens within a short
+time, try again with the browser's web console open. Sometimes you will see
+403 errors and other messages that might indicate you don't have the correct
+details set up.
```
Comment (zwuis) [inline llvm/docs/HowToAddABuilder.rst:307]: nit: new line character at eof

Comment (DavidSpickett) [reply llvm/docs/HowToAddABuilder.rst]: Done.

**Comment by DavidSpickett** [comment]
ping! @gkistanova

**Comment by DavidSpickett** [comment]
ping!

Though I know the instructions do work, so I will go ahead and land this if there's no input in the next week.

**Comment by asb** [comment]
Firstly, thank you for documenting this - I recently had to reach out to Galina for help with this, but I failed to think to follow up to document it.

I know it's post-commit, but two thoughts:
* Might it be better to suggest people go to to https://github.com/settings/emails while logged in to review the emails associated with their account? For me at least, github.com/asb doesn't show anything very useful.
* I was surprised to hear it's working for you with an email other than your primary one. I wasn't able to get login to work on the lab.llvm.org until I switched the admin email for my bots over to asb@asbradbury.org (my GH primary email) rather than asb@igalia.com (the backup email). It's possible there was something else going on (e.g. caching) that caused me to think that was the fix.

**Comment by DavidSpickett** [comment]
The key point here is that Buildbot sees the email listed on your public GitHub profile page. That email must therefore be associated with the account, but it can be primary or secondary, as long as you've chosen to show it on your profile.

I see you have `asb@asbradbury.org` on your public profile, so by switching to that now Buildbot sees it on your public profile as well and you're seen as the admin.

This is what had me stuck for ages, it was only by chance a colleague interpreted "GitHub email" to mean "GitHub profile email" and we realised what was happening.

That said, if that's not clear I'll update the wording.

**Comment by asb** [comment]
> The key point here is that Buildbot sees the email listed on your public GitHub profile page. That email must therefore be associated with the account, but it can be primary or secondary, as long as you've chosen to show it on your profile.

You know I think this is the key reason for my confusion, because when you login and grant oauth permissions to the buildbot it requests permission to "Access user email addresses (read-only)". ie. the user:email permission [here](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps) which should allow listing all email addresses https://docs.github.com/en/rest/users/emails?apiVersion=2022-11-28#list-email-addresses-for-the-authenticated-user

**Pull request merged**